datacube/virtual/impl.py

```diff
@@ -824,8 +827,8 @@ def fetch(self, grouped: VirtualDatasetBox, **load_settings: Dict[str, Any]) ->
         result = xarray.Dataset()
         result.coords['time'] = grouped.box.coords['time']
 
-        for name, coord in grouped.geobox.coordinates.items():
-            result.coords[name] = (name, coord.values, {'units': coord.units, 'resolution': coord.resolution})
+        coords = OrderedDict(**geobox.xr_coords(with_crs=spatial_ref))
+        result.coords.update(coords)
```

Comment:
should just mention that these two lines can be collapsed into:
```
result.coords.update(geobox.xr_coords(with_crs='spatial_ref'))
```

Reply:
in which case the collections import won't be needed.

Reply:
I've copied those lines from `create_storage` in datacube.api, not sure if it's still doing the right thing if omitting the `OrderedDict`...